test/test_stream.jl

```diff
@@ -12,14 +12,10 @@ buf = IOBuffer()
 pipe = Base64EncodePipe(buf)
 write_to_png(c,pipe)
 close(pipe)
+
 # Catch short writes
 
-    if VERSION >= v"0.6.0-dev.1954"
-        str = String(take!(buf))
-        str_data = Vector{UInt8}(str)
-    else
-        str = takebuf_string(buf)    
-        str_data = str.data
-    end        
+@compat str = String(take!(buf))
```

Comment:
you don't actually need the `@compat` macro for this since it isn't a syntax rewrite, these functions are just exported by Compat